### PR TITLE
Jusjohns xfcontrols cardview

### PR DESCRIPTION
**Pull request by profexorgeek**
This PR adds a sample that demonstrates how to create a custom control by extending the `ContentView`.

### CODE REVIEW

File: UserInterface/CardViewDemo/CardViewDemo/CardViewDemo.Android/Properties/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android" android:versionCode="1" android:versionName="1.0" package="com.companyname.cardviewdemo">
+    <uses-sdk android:minSdkVersion="21" android:targetSdkVersion="28" />
```
Comment (davidbritch) [inline UserInterface/CardViewDemo/CardViewDemo/CardViewDemo.Android/Properties/AndroidManifest.xml]: min should be 19.

Remove targetSdkVersion.

File: UserInterface/CardViewDemo/CardViewDemo/CardViewDemo/ControlTemplatePage.xaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    <ContentPage.Resources>
+        <ResourceDictionary>
+            <ControlTemplate x:Key="CardViewCompressed">
+                <Grid>
+                    <Grid.RowDefinitions>
+                        <RowDefinition Height="100" />
+                    </Grid.RowDefinitions>
+                    <Grid.ColumnDefinitions>
+                        <ColumnDefinition Width="100" />
+                        <ColumnDefinition Width="100*" />
```
Comment (davidbritch) [inline UserInterface/CardViewDemo/CardViewDemo/CardViewDemo/ControlTemplatePage.xaml]: Does the asterisk after the number have any effect? If you just want the second column to take the remaining space just use *.

File: UserInterface/CardViewDemo/CardViewDemo/CardViewDemo/ControlTemplatePage.xaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                    </Grid.ColumnDefinitions>
+                    <Image Grid.Row="0"
```
Comment (davidbritch) [inline UserInterface/CardViewDemo/CardViewDemo/CardViewDemo/ControlTemplatePage.xaml]: Grid.Row="0" and Grid.Column="0" are redundant here. You don't need to specify either for the zeroth row/column. Consequently, you only have to specify them for row or column > 0.

File: UserInterface/CardViewDemo/CardViewDemo/CardViewDemo/ControlTemplatePage.xaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                           Grid.Column="0"
+                           Source="{TemplateBinding IconImageSource}"
+                           BackgroundColor="{TemplateBinding IconBackgroundColor}"
+                           WidthRequest="100"
+                           HeightRequest="100"
+                           Aspect="AspectFill"
+                           HorizontalOptions="Center"
+                           VerticalOptions="Center" />
+                    <StackLayout Grid.Row="0"
```
Comment (davidbritch) [inline UserInterface/CardViewDemo/CardViewDemo/CardViewDemo/ControlTemplatePage.xaml]: As above.

File: UserInterface/CardViewDemo/CardViewDemo/CardViewDemo/ControlTemplatePage.xaml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        <Label Text="A standard CardView control is suitable for grid layouts:"
+               FontAttributes="Italic"
+               TextColor="Red"
+               Margin="0,0,0,10" />
+        <controls:CardView BorderColor="DarkGray"
+                           CardTitle="Slavko Vlasic"
+                           CardDescription="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla elit dolor, convallis non interdum."
+                           IconBackgroundColor="SlateGray"
+                           IconImageSource="user.png" />
+        <Label Text="A ControlTemplate overrides standard view, creating a more compact view suitable for lists:"
```
Comment (davidbritch) [inline UserInterface/CardViewDemo/CardViewDemo/CardViewDemo/ControlTemplatePage.xaml]: A ControlTemplate overrides **the** standard view...

File: UserInterface/CardViewDemo/CardViewDemo/CardViewDemo/Controls/CardView.xaml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+                <RowDefinition Height="Auto" />
+            </Grid.RowDefinitions>
+            <Grid.ColumnDefinitions>
+                <ColumnDefinition Width="75" />
+                <ColumnDefinition Width="200" />
+            </Grid.ColumnDefinitions>
+            <Frame Grid.Row="0"
+                   Grid.Column="0"
+                   IsClippedToBounds="True"
+                   BindingContext="{x:Reference this}"
```
Comment (davidbritch) [inline UserInterface/CardViewDemo/CardViewDemo/CardViewDemo/Controls/CardView.xaml]: You only need to set the BindingContext on the root Frame, as it gets inherited by all children in the visual tree. So just set on the root Frame, and remove it from every other control.

File: UserInterface/CardViewDemo/CardViewDemo/CardViewDemo/ViewModels/PersonViewModel.cs [hunk 7]
```diff
@@ -0,0 +1,57 @@
+﻿using System.ComponentModel;
+using System.Runtime.CompilerServices;
+
+namespace CardViewDemo.ViewModels
+{
+    public class PersonViewModel : INotifyPropertyChanged
+    {
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        private string photo;
```
Comment (davidbritch) [inline UserInterface/CardViewDemo/CardViewDemo/CardViewDemo/ViewModels/PersonViewModel.cs]: Remove private throughout class.

File: UserInterface/CardViewDemo/CardViewDemo/CardViewDemo/ControlTemplatePage.xaml (hunk 2 above)
Comment (profexorgeek) [reply UserInterface/CardViewDemo/CardViewDemo/CardViewDemo/ControlTemplatePage.xaml]: This came from a misunderstanding of the asterisk and worked for the wrong reasons. Fixed.

**Pull request merged**